core/src/main/scala/kafka/controller/ControllerContext.scala

```diff
@@ -347,6 +362,7 @@ class ControllerContext {
         partitionLeadershipInfo.remove(new TopicPartition(topic, partition))
       }
     }
+    partitionAssignmentsChanged = true
```

Comment:
What about setting this in clearTopicsState() as well? It wipes out partitionAssignments...is there a corner case there? It feels like it, but I don't know under what conditions it's called.

Reply:
Good catch. I missed that case.